Error with `add-jwt-key` target within the `local-env-setup` make target

When running (latest `main`) the `make local-env-setup` I seem to run into an issue with the `add-jwt-key` target. Below is a bit of the log:

```
kubectl patch deployment mcp-broker-router -n mcp-system --patch-file config/mcp-gateway/overlays/mcp-system/poll-interval-patch.yaml
deployment.apps/mcp-broker-router patched
make[1]: Leaving directory '/home/matzew/go/src/github.com/kagenti/mcp-gateway'
"make" add-jwt-key
make[1]: Entering directory '/home/matzew/go/src/github.com/kagenti/mcp-gateway'
Error from server (Conflict): error when applying patch:
{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"apps/v1\",\"kind\":\"Deployment\",\"metadata\":{\"annotations\":{\"deployment.kubernetes.io/revision\":\"2\"},\"creationTimestamp\":\"2026-02-11T13:48:29Z\",\"generation\":2,\"labels\":{\"app\":\"mcp-broker-router\",\"component\":\"broker-router\"},\"name\":\"mcp-broker-router\",\"namespace\":\"mcp-system\",\"resourceVersion\":\"1075\",\"uid\":\"4e9c2ff1-928f-482f-9c3a-15cbc43936b4\"},\"spec\":{\"progressDeadlineSeconds\":600,\"replicas\":1,\"revisionHistoryLimit\":10,\"selector\":{\"matchLabels\":{\"app\":\"mcp-broker-router\",\"component\":\"broker-router\"}},\"strategy\":{\"rollingUpdate\":{\"maxSurge\":\"25%\",\"maxUnavailable\":\"25%\"},\"type\":\"RollingUpdate\"},\"template\":{\"metadata\":{\"creationTimestamp\":null,\"labels\":{\"app\":\"mcp-broker-router\",\"component\":\"broker-router\"}},\"spec\":{\"containers\":[{\"command\":[\"./mcp_gateway\",\"--mcp-gateway-config=/config/config.yaml\",\"--mcp-broker-public-address=0.0.0.0:8080\",\"--mcp-gateway-public-host=mcp.127-0-0-1.sslip.io\",\"--mcp-router-address=0.0.0.0:50051\",\"--mcp-check-interval=10\",\"--log-level=-4\"],\"env\":[{\"name\":\"MCP_GATEWAY_PUBLIC_HOST\",\"value\":\"mcp.127-0-0-1.sslip.io\"},{\"name\":\"TRUSTED_HEADER_PUBLIC_KEY\",\"valueFrom\":{\"secretKeyRef\":{\"key\":\"key\",\"name\":\"trusted-headers-public-key\"}}}],\"image\":\"ghcr.io/kuadrant/mcp-gateway:latest\",\"imagePullPolicy\":\"IfNotPresent\",\"livenessProbe\":{\"failureThreshold\":3,\"httpGet\":{\"path\":\"/healthz\",\"port\":8080,\"scheme\":\"HTTP\"},\"initialDelaySeconds\":10,\"periodSeconds\":30,\"successThreshold\":1,\"timeoutSeconds\":1},\"name\":\"mcp-broker-router\",\"ports\":[{\"containerPort\":8080,\"name\":\"http\",\"protocol\":\"TCP\"},{\"containerPort\":8181,\"name\":\"config\",\"protocol\":\"TCP\"},{\"containerPort\":50051,\"name\":\"grpc\",\"protocol\":\"TCP\"}],\"readinessProbe\":{\"failureThreshold\":3,\"httpGet\":{\"path\":\"/readyz\",\"port\":8080,\"scheme\":\"HTTP\"},\"initialDelaySeconds\":5,\"periodSeconds\":10,\"successThreshold\":1,\"timeoutSeconds\":1},\"resources\":{\"limits\":{\"cpu\":\"500m\",\"memory\":\"512Mi\"},\"requests\":{\"cpu\":\"100m\",\"memory\":\"128Mi\"}},\"terminationMessagePath\":\"/dev/termination-log\",\"terminationMessagePolicy\":\"File\",\"volumeMounts\":[{\"mountPath\":\"/config\",\"name\":\"config-volume\"}]}],\"dnsPolicy\":\"ClusterFirst\",\"restartPolicy\":\"Always\",\"schedulerName\":\"default-scheduler\",\"securityContext\":{},\"serviceAccount\":\"mcp-broker-router\",\"serviceAccountName\":\"mcp-broker-router\",\"terminationGracePeriodSeconds\":30,\"volumes\":[{\"name\":\"config-volume\",\"secret\":{\"defaultMode\":420,\"secretName\":\"mcp-gateway-config\"}}]}}},\"status\":{\"conditions\":[{\"lastTransitionTime\":\"2026-02-11T13:48:29Z\",\"lastUpdateTime\":\"2026-02-11T13:48:29Z\",\"message\":\"Deployment does not have minimum availability.\",\"reason\":\"MinimumReplicasUnavailable\",\"status\":\"False\",\"type\":\"Available\"},{\"lastTransitionTime\":\"2026-02-11T13:48:29Z\",\"lastUpdateTime\":\"2026-02-11T13:48:29Z\",\"message\":\"Created new replica set \\\"mcp-broker-router-bbb4569cd\\\"\",\"reason\":\"NewReplicaSetCreated\",\"status\":\"True\",\"type\":\"Progressing\"}],\"observedGeneration\":1,\"replicas\":1,\"unavailableReplicas\":1,\"updatedReplicas\":1}}\n"},"resourceVersion":"1075"},"spec":{"template":{"spec":{"$setElementOrder/containers":[{"name":"mcp-broker-router"}],"containers":[{"$setElementOrder/env":[{"name":"MCP_GATEWAY_PUBLIC_HOST"},{"name":"TRUSTED_HEADER_PUBLIC_KEY"}],"env":[{"name":"TRUSTED_HEADER_PUBLIC_KEY","valueFrom":{"secretKeyRef":{"key":"key","name":"trusted-headers-public-key"}}}],"name":"mcp-broker-router"}]}}},"status":{"$setElementOrder/conditions":[{"type":"Available"},{"type":"Progressing"}],"conditions":[{"message":"Created new replica set \"mcp-broker-router-bbb4569cd\"","reason":"NewReplicaSetCreated","type":"Progressing"}],"observedGeneration":1,"replicas":1,"unavailableReplicas":1}}
to:
Resource: "apps/v1, Resource=deployments", GroupVersionKind: "apps/v1, Kind=Deployment"
Name: "mcp-broker-router", Namespace: "mcp-system"
for: "STDIN": error when patching "STDIN": Operation cannot be fulfilled on deployments.apps "mcp-broker-router": the object has been modified; please apply your changes to the latest version and try again
make[1]: *** [Makefile:501: add-jwt-key] Error 1
make[1]: Leaving directory '/home/matzew/go/src/github.com/kagenti/mcp-gateway'
make: *** [Makefile:477: local-env-setup] Error 2
```


used to work with older version - did update on `main` and saw this

--- a/Makefile
+++ b/Makefile
@@ -498,9 +498,11 @@ local-env-teardown: ## Tear down the local Kind cluster
 
 .PHONY: add-jwt-key
 add-jwt-key: #add the public key needed to validate any incoming jwt based headers such as x-allowed-tools
-	@kubectl get deployment/mcp-broker-router -n mcp-system -o yaml | \
+	@for i in 1 2 3 4 5; do \
+		kubectl get deployment/mcp-broker-router -n mcp-system -o yaml | \
 		bin/yq '.spec.template.spec.containers[0].env += [{"name":"TRUSTED_HEADER_PUBLIC_KEY","valueFrom":{"secretKeyRef":{"name":"trusted-headers-public-key","key":"key"}}}] | .spec.template.spec.containers[0].env |= unique_by(.name)' | \
-		kubectl apply -f -
+		kubectl apply -f - && break || (echo "Retry $$i/5 failed, waiting 2s..." && sleep 2); \
+	done
 
 .PHONY: dev
 dev: ## Setup cluster for local development (binaries run on host)

